### PR TITLE
Add RSS for news

### PR DESCRIPTION
**Pull request by nanaya**
Opengraph related stuff will be added later.

### CODE REVIEW

File: resources/views/news/index-rss.blade.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                <title>{{ $post->title() }}</title>
+                <link>{{ route('news.show', $post->slug) }}</link>
+                <guid isPermaLink="false">{{ atom_id('news', $post->getKey()) }}</guid>
+                <pubDate>{{ $post->published_at->toRfc7231String() }}</pubDate>
+                <description>{!! $post->previewText() !!}</description>
+                @if ($post->firstImage() !== null)
+                    <enclosure
+                        url="{{ $post->firstImage(true) }}"
+                        length="300"
+                        type="image/jpeg"
```
Comment (cl8n) [inline resources/views/news/index-rss.blade.php]: should try to infer the type, these aren't guaranteed to be jpeg (looks like already a few are png and one is gif)

Comment (nanaya) [reply resources/views/news/index-rss.blade.php]: As it turned out it's not even useful.

File: app/Models/NewsPost.php [hunk 2]
```diff
@@ -204,9 +204,23 @@ public function editUrl()
         return 'https://github.com/'.OsuWiki::USER.'/'.OsuWiki::REPOSITORY.'/tree/master/news/'.$this->filename();
     }
 
-    public function firstImage()
+    public function firstImage($absolute = false)
```
Comment (nanaya) [inline app/Models/NewsPost.php]: as turned out this isn't needed for this PR but it'll be needed for opengraph stuff, so... ┐(°～° )┌

**Comment by camellirite** [comment]
this is neat but is there a link to it that im missing? i want to have a neat little newsfeed on my desktop using rainmeter

**Comment by LiquidPL** [comment]
https://osu.ppy.sh/home/news?format=rss

**Pull request merged**